### ISSUE TITLE
Proper fix for dirname in env.sh

### ISSUE THREAD

**Issue by hohwille** [devonfw/devon-ide#23]
User @maihacke changed `env.sh` to use `$(dirname "$0")`:
https://github.com/devonfw/devon-ide/commit/b9cca7862995eafbac07000f55eafa5d639019c7#diff-59bbdf4f0007715da1871587ba4ffd3d

I assume this was some kind of bugfix for MacOS. Also this seems more reliable to me as `$BASH_SOURCE` is proprietary and only works in bash. What is the purpose why this was ever changed to this?

User @vapadwal wants to change `env.sh` back to `$(dirname "$BASH_SOURCE")`:
https://github.com/devonfw/devon-ide/commit/985a51ed992edd9d129d963c61e25db6460ba357#diff-59bbdf4f0007715da1871587ba4ffd3d

As I do not like to have git wars, I would kindly ask all participants to clarify the reasons for these changes and find a peaceful solution.

**Comment by hohwille** [comment]
I was able to reproduce the error somehow:
```
Jorgs-MacBook-Pro:devon hohwille$ . env.sh
dirname: illegal option -- b
usage: dirname path
IDE environment has been initialized.
Jorgs-MacBook-Pro:devon hohwille$ echo $0
-bash
```

So in some cases bash has `-bash` as `$0` what is causing the issue.
Hence I think both "fixes" are wrong.
IMHO we need an if that tests if we are in bash and then uses `$BASH_SOURCE` and otherwise uses `$0` to make it work in any shell.

**Comment by maihacke** [comment]
Please try 
. ./env.sh

**Comment by hohwille** [comment]
Create `test.sh` with this content:
```
echo "0=$0"
echo "BASH_SOURCE=$BASH_SOURCE"
echo "_=$_"
````

This will produce:
```
$ . ./test.sh
0=-bash
BASH_SOURCE=./test.sh
_=BASH_SOURCE=./test.sh
```

**Comment by hohwille** [comment]
```
$ zsh
% . ./test.sh
0=./test.sh
BASH_SOURCE=
_=BASH_SOURCE=
```

So in `zsh` the `$0` variant is correct.

**Comment by hohwille** [comment]
`$_` seems to be the last command (even within the current script)

**Comment by maihacke** [comment]
yepp, nonsense :-/

**Comment by maihacke** [comment]
maybe we should go with ${BASH_SOURCE[0]-$0} then
Should work in Bash and ZSH

**Comment by hohwille** [comment]
We only need to review and merge PR #25 that fixes this issue together with #21.

**Comment by hohwille** [comment]
fixed with PR #21
